src/components/Identity/setCustomerIds.js

```diff
@@ -10,30 +8,14 @@ const makeServerCall = (payload, lifecycle, network) => {
 };
 
 export default (ids, cookieJar, lifecycle, network, optIn) => {
-  validateCustomerIds(ids);
   const event = createEvent(); // FIXME: We shouldn't need an event.
   event.mergeData({}); // FIXME: We shouldn't need an event.
   const payload = network.createPayload();
   payload.addEvent(event); // FIXME: We shouldn't need an event.
-  const customerIds = assign({}, ids);
-  const customerIdsProcess = processCustomerIds(customerIds);
-  const customerIdChanged = customerIdsProcess.detectCustomerIdChange(
-    cookieJar
-  );
-  customerIdsProcess
-    .getNormalizedAndHashedIds()
-    .then(normalizedAndHashedIds => {
-      const idNames = Object.keys(normalizedAndHashedIds);
-      idNames.forEach(idName => {
-        payload.addIdentity(idName, normalizedAndHashedIds[idName]);
-      });
-      payload.mergeMeta({ identity: { customerIdChanged } });
-      if (customerIdChanged) {
-        customerIdsProcess.updateChecksum(cookieJar);
-      }
-      return lifecycle
-        .onBeforeEvent(event, {}, false) // FIXME: We shouldn't need an event.
-        .then(() => optIn.whenOptedIn())
-        .then(() => makeServerCall(payload, lifecycle, network));
-    });
+  addCustomerIdsToPayload(ids, cookieJar, payload).then(() => {
```

Comment:
I believe we should be returning this promise chain for a couple reasons: (1) When a customer executes `setCustomerIds`, I think it would be appropriate to resolve the promise returned to the customer _after_ the event has been sent to the server and (2) if an error is thrown anywhere in this promise chain, we want to reject the promise returned to the customer.